### PR TITLE
Run, RunAsync and  Iter return Unit

### LINKED ISSUE #28: Make all Iter, Run and RunAsync return Unit
It was returning the original entity out of convenience, but in truth is a terrible practice.

### PR DESCRIPTION
**Pull request by theonlyroboboro**
closes #28

### CODE REVIEW

File: src/DeFuncto.Core/Types/AsyncOption.cs [hunk 1]
```diff
@@ -123,59 +123,59 @@ namespace DeFuncto
             Option.Map(opt => opt.Where(filter));
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public Task<Option<T>> Iter(Action<T> f) =>
-            Option.Run(opt => opt.Iter(f));
+        public Task<Option<Unit>> Iter(Action<T> f) =>
```
Comment (JYCabello) [inline src/DeFuncto.Core/Types/AsyncOption.cs:126]: Task<Unit>

File: src/DeFuncto.Tests/Core/Extensions/Objects/Run.cs [hunk 2]
```diff
@@ -11,9 +11,8 @@ public void Runs()
         {
             var witness = new Witness();
             witness
-                .Run(w => w.Touch())
-                .Run(w => { w.Touch(); });
-            Assert.Equal(2, witness.TimesCalled);
+                .Run(w => w.Touch());
```
Comment (JYCabello) [inline src/DeFuncto.Tests/Core/Extensions/Objects/Run.cs:14]: Reintroduce the other .Run, they are different overloads and both were tested

**Review by JYCabello — CHANGES_REQUESTED** [review]
It's also for `Result` and `AsyncResult`

**Review by JYCabello — CHANGES_REQUESTED** [review]
It's still missing the `Iter` in `Result` and `AsyncResult`

**Comment by theonlyroboboro** [comment]
@JYCabello fixt

**Review by JYCabello — APPROVED** [review]

**Pull request merged**